emark/utils.py

```diff
@@ -122,3 +126,23 @@ def __str__(self) -> str:
         # sanitize all wide vertical or horizontal spaces
         text = self.DOUBLE_NEWLINE.sub("\n\n", text.strip())
         return self.DOUBLE_SPACE.sub(" ", text)
+
+
+def extract_domain(url: str) -> str:
+    """Extracts the registered domain from a given URL.
+
+    If the domain is "localhost", it includes the port number in the returned string.
+
+    Args:
+        url (str): The URL from which to extract the domain.
+
+    Returns:
+        str: The registered domain or "localhost" with port if applicable.
+    """
+    extractor = tldextract.TLDExtract(suffix_list_urls=())
+    extracted = extractor(url)
+    if extracted.domain == "localhost":
+        if port := parse.urlparse(url).port:
+            return f"{extracted.domain}:{port}"
+        return extracted.domain
```

Comment:
The port part shouldn't only apply to localhost. Any domain may include a port.
```suggestion
      if port := parse.urlparse(url).port:
          return f"{extracted.domain}:{port}"
      return extracted.domain
```